src/Polly.Core/ResilienceProperties.cs

```diff
@@ -20,10 +20,18 @@ public sealed class ResilienceProperties
     /// <returns>True, if a property was retrieved.</returns>
     public bool TryGetValue<TValue>(ResiliencePropertyKey<TValue> key, [MaybeNullWhen(false)] out TValue value)
     {
-        if (Options.TryGetValue(key.Key, out object? val) && val is TValue typedValue)
+        if (Options.TryGetValue(key.Key, out object? val))
         {
-            value = typedValue;
-            return true;
+            if (val is TValue typedValue)
+            {
+                value = typedValue;
+                return true;
+            }
+            else if (val is null)
+            {
+                value = default!;
```

Comment:
The `!` is wrong here, if `null` is a valid value.

Reply:
Added a comment describing the need to use `!` operator.